apps/go-mod-validator/README.md

```diff
@@ -0,0 +1,30 @@
+# go-mod-validator
+
+This action verifies that all upstream dependencies are tested in a given go
+package.
+
+Requirements:
+
+1. Go compiler (version matching go.mod or higher)
+2. jq - https://jqlang.github.io/jq/ (available by default in most linux
+   distributions)
+
+Validations implemented so far,
+
+1. Finds all `go.mod` files recursively and verifies dependencies are present in
+   the default branch of the upstream dependency's repositories
+
+## Example usage
+
+```yaml
+steps:
+  - name: Check out the repository
+    uses: actions/checkout
+  - name: Setup go
+    uses: actions/setup-go
+  - name: go-mod-validation
+    uses: smartcontractkit/.github/apps/go-mod-validator
```

Comment:
```suggestion
  - name: Validate go.mod
    uses: smartcontractkit/.github/apps/go-mod-validator@<commit> # go-mod-validator@x.y.z
```

We recommend tagging our monorepo actions with a commit and a commented tag so I typically try to make people who might copy paste, to actually grab a specific version by doing this ([example](https://github.com/smartcontractkit/.github/blob/main/actions/policy-bot-config-validator/README.md)).